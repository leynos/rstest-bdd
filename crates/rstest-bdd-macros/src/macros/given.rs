//! Implementation of the `#[given]` macro.

use proc_macro::TokenStream;

/// Macro for defining a Given step that registers with the step inventory.
pub(crate) fn given(attr: TokenStream, item: TokenStream) -> TokenStream {
    super::step_attr(attr, item, crate::StepKeyword::Given)
}
