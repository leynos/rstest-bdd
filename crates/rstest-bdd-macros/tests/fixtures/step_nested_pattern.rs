use rstest_bdd_macros::given;

struct User {
    coords: (i32, i32),
}

#[given("user coords")]
fn step_nested(User { coords: (x, y) }: User) {}

fn main() {}
