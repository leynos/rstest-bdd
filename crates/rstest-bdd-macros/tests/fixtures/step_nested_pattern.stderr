error: complex destructuring patterns are not yet supported - pattern resolves to 2 identifiers but a single bare identifier is required. Found pattern: `User { coords : (x, y) }`
 --> tests/fixtures/step_nested_pattern.rs:8:16
  |
8 | fn step_nested(User { coords: (x, y) }: User) {}
  |                ^^^^^^^^^^^^^^^^^^^^^^^
